### ISSUE TITLE
union/union all bug wrapped queries 

### ISSUE THREAD

**Issue by fogine** [knex/knex#5047]
# Environment

Knex version: 1.0.3
Database + version: PostgreSQL 13.6
OS: Linux

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do
The following simple query:

```javascript
    const nextPrevEarningsQ = knex('table').select('date')
    .where('date', '>', date).orderBy('date').limit(1).unionAll(function() {
        this.select('date').from('table')
            .where('date', '<', date).orderBy('date', 'desc').limit(1);
    }, true);
```

generates incorrect query:

```sql
select "date" from "earnings_calendar" where "date" > '2022-01-03' union all (select "date" from "earnings_calendar" where "date" < '2022-01-03' order by "date" desc limit 1) order by "date" asc limit 1;
```

which returns results only from second wrapped query.

it should generate 

```sql
(select "date" as next from "earnings_calendar" where "date" > '2022-01-03' order by next asc limit 1) union all (select date as prev from "earnings_calendar" where "date" < '2022-01-03' order by prev desc limit 1);
```

which would give correct results from both queries..

**Comment by OlivierCavadenti** [comment]
I have a work in progress here : https://github.com/knex/knex/pull/5030
It works for PostgreSQL but this syntax doesnt work for all SGBD, so I still need some work (I plan to finish this PR this weekend).

**Comment by kibertoad** [comment]
Released in 1.0.4
